chia/full_node/mempool_manager.py

```diff
@@ -540,27 +562,27 @@ async def new_peak(
             if last_npc_result.conds is not None:
                 for spend in last_npc_result.conds.spends:
                     if spend.coin_id in self.mempool.removals:
-                        item = self.mempool.removals[bytes32(spend.coin_id)]
-                        self.mempool.remove_from_pool(item)
-                        self.remove_seen(item.spend_bundle_name)
+                        c_ids: List[bytes32] = self.mempool.removals[bytes32(spend.coin_id)]
+                        self.mempool.remove_from_pool(c_ids)
+                        for c_id in c_ids:
+                            self.remove_seen(c_id)
         else:
             old_pool = self.mempool
             self.mempool = Mempool(self.mempool_max_total_cost)
+            self.seen_bundle_hashes = {}
             for item in old_pool.spends.values():
-                _, result, _ = await self.add_spendbundle(
-                    item.spend_bundle, item.npc_result, item.spend_bundle_name, item.program
-                )
+                _, result, _ = await self.add_spend_bundle(item.spend_bundle, item.npc_result, item.spend_bundle_name)
                 # If the spend bundle was confirmed or conflicting (can no longer be in mempool), it won't be
                 # successfully added to the new mempool. In this case, remove it from seen, so in the case of a reorg,
                 # it can be resubmitted
-                if result != MempoolInclusionStatus.SUCCESS:
-                    self.remove_seen(item.spend_bundle_name)
+                if result == MempoolInclusionStatus.SUCCESS:
+                    self.seen(item.spend_bundle_name)
```

Comment:
why do we call this method here? it only checks if spend bundle has been seen recently so kind of no-op

Reply:
It avoids re-processing transactions that we have already processed.

Reply:
but `self.seen()` doesn't change any state? there is a `add_and_maybe_pop_seen` but it's never used I think